### PR TITLE
Created UG documents for state changing event for checkbox and badge visibility support for BadgeView.

### PR DESCRIPTION
**Pull request by KarthikaRavichandranSF4617**

### CODE REVIEW

File: MAUI/Badge-View/Badge-customization.md [hunk 1]
```diff
@@ -344,3 +344,46 @@ Content = sfBadgeView;
 {% endtabs %}
 
 ![BadgeView Alignment](badge-customization_images/badge_alignment.png)
+
+## Visibility of badge
+
+The IsVisible property is used to show or hide the badge in the BadgeView. By default, visibility of the badge is enabled.
+
+N> The badge appears only when the `IsVisible` property value is set to **true**.
```
Comment (naveenkumar-sanjeevirayan) [inline MAUI/Badge-View/Badge-customization.md:352]: @KarthikaRavichandranSF4617, its not needed, you can remove it.

File: MAUI/Badge-View/Badge-customization.md [hunk 2]
```diff
@@ -344,3 +344,46 @@ Content = sfBadgeView;
 {% endtabs %}
 
 ![BadgeView Alignment](badge-customization_images/badge_alignment.png)
+
+## Visibility of badge
+
+The IsVisible property is used to show or hide the badge in the BadgeView. By default, visibility of the badge is enabled.
```
Comment (naveenkumar-sanjeevirayan) [inline MAUI/Badge-View/Badge-customization.md:350]: You can show or hide the badges in the BadgeView by setting the IsVisible property in the BadgeSetting. By default, the badge will be visible.

File: MAUI/Badge-View/Badge-customization.md [hunk 3]
```diff
@@ -344,3 +344,46 @@ Content = sfBadgeView;
...
+
+N> The badge appears only when the `IsVisible` property value is set to **true**.
+
+{% tabs %}
+{% highlight xaml %}
+
+<sf:SfBadgeView BadgeText="20">
```
Comment (naveenkumar-sanjeevirayan) [inline MAUI/Badge-View/Badge-customization.md:357]: use proper namespace like below,

<badge:SfBadgeView BadgeText="20">

File: MAUI/Badge-View/Badge-customization.md [hunk 4]
```diff
@@ -344,3 +344,46 @@ Content = sfBadgeView;
...
+badgeView.BadgeSettings=badgeSettings;
+badgeVuew.Content=button;
+Content=badgeView;
+
+{% endhighlight %}
+{% endtabs %}
+
+If IsVisible is true
+
+![.NET MAUI BadgeView with Badge](badge-customization_images/net_maui_badgeview_with_badge.png)
```
Comment (naveenkumar-sanjeevirayan) [inline MAUI/Badge-View/Badge-customization.md:385]: Add two badge view images in the single image

File: MAUI/CheckBox/Events.md [hunk 5]
```diff
@@ -55,4 +55,38 @@ This event occurs when the value or state of the [`IsChecked`](https://help.sync
...
+
+The StateChanging event occurs when the state of [`IsChecked`](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Buttons.SfCheckBox.html#Syncfusion_Maui_Buttons_SfCheckBox_IsChecked) property is about to change in the Checkbox control. The StateChangingEventArgs provides the following properties:
+
+* [`IsChecked`](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Buttons.SfCheckBox.html#Syncfusion_Maui_Buttons_SfCheckBox_IsChecked) : The new value(state) of the [`IsChecked`](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Buttons.SfCheckBox.html#Syncfusion_Maui_Buttons_SfCheckBox_IsChecked) property.
+* `Cancel` : Gets or sets a value indicating whether the event should be canceled.
+
+{% tabs %}
+{% highlight xaml %}
+
+    <syncfusion:SfCheckBox x:Name="checkBox" Text="Unchecked State" StateChanging="CheckBox_StateChanging"/>
```
Comment (naveenkumar-sanjeevirayan) [inline MAUI/CheckBox/Events.md:70]: Set proper text to the check box control.

File: MAUI/CheckBox/Events.md [hunk 6]
```diff
@@ -55,4 +55,38 @@ This event occurs when the value or state of the [`IsChecked`](https://help.sync
 {% endhighlight %}
 {% endtabs %}
 
-![.NET MAUI CheckBox](Images/Getting-Started/tristate.png)
\ No newline at end of file
+![.NET MAUI CheckBox](Images/Getting-Started/tristate.png)
+
+## StateChanging event
+
+The StateChanging event occurs when the state of [`IsChecked`](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Buttons.SfCheckBox.html#Syncfusion_Maui_Buttons_SfCheckBox_IsChecked) property is about to change in the Checkbox control. The StateChangingEventArgs provides the following properties:
```
Comment (naveenkumar-sanjeevirayan) [inline MAUI/CheckBox/Events.md:62]: We trigger it, when we tap on the control alone. So, need to change content for this.

Comment (KarthikaRavichandranSF4617) [reply MAUI/CheckBox/Events.md]: I have changed the content for this.

File: MAUI/CheckBox/Events.md (hunk 5 above)
Comment (KarthikaRavichandranSF4617) [reply MAUI/CheckBox/Events.md]: I have updated the text to the checkbox control.

File: MAUI/Badge-View/Badge-customization.md (hunk 3 above)
Comment (KarthikaRavichandranSF4617) [reply MAUI/Badge-View/Badge-customization.md]: Updated the namespace.

File: MAUI/Badge-View/Badge-customization.md (hunk 2 above)
Comment (KarthikaRavichandranSF4617) [reply MAUI/Badge-View/Badge-customization.md]: Updated the description.

File: MAUI/Badge-View/Badge-customization.md (hunk 1 above)
Comment (KarthikaRavichandranSF4617) [reply MAUI/Badge-View/Badge-customization.md]: I have removed it.

File: MAUI/Badge-View/Badge-customization.md [hunk 7]
```diff
@@ -344,3 +344,38 @@ Content = sfBadgeView;
 {% endtabs %}
 
 ![BadgeView Alignment](badge-customization_images/badge_alignment.png)
+
+## Visibility of badge
+
+You can show or hide the badges in the BadgeView by setting the IsVisible property in the BadgeSetting. By default, the badge will be visible.
```
Comment (naveenkumar-sanjeevirayan) [inline MAUI/Badge-View/Badge-customization.md:350]: You can show or hide the badges in the .NET MAUI BadgeView by setting the `IsVisible` property in the BadgeSetting. By default, the badge will be visible.

File: MAUI/Badge-View/Badge-customization.md [hunk 8]
```diff
@@ -344,3 +344,38 @@ Content = sfBadgeView;
...
+
+{% tabs %}
+{% highlight xaml %}
+
+<badge:SfBadgeView BadgeText="20">
+    <badge:SfBadgeView.Content>
+        <Button Text="Message"></Button>
```
Comment (naveenkumar-sanjeevirayan) [inline MAUI/Badge-View/Badge-customization.md:357]: <Button Text="Message"/>

File: MAUI/CheckBox/Events.md [hunk 9]
```diff
@@ -55,4 +55,38 @@ This event occurs when the value or state of the [`IsChecked`](https://help.sync
 {% endhighlight %}
 {% endtabs %}
 
-![.NET MAUI CheckBox](Images/Getting-Started/tristate.png)
\ No newline at end of file
+![.NET MAUI CheckBox](Images/Getting-Started/tristate.png)
+
+## StateChanging event
+
+The StateChanging event is triggered when the CheckBox control is tapped, indicating that the state of the [`IsChecked`](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Buttons.SfCheckBox.html#Syncfusion_Maui_Buttons_SfCheckBox_IsChecked) property in the Checkbox control is about to change. The StateChangingEventArgs provides the following properties:
```
Comment (naveenkumar-sanjeevirayan) [inline MAUI/CheckBox/Events.md:62]: Highlight the event and class names like below,

The `StateChanging` event is triggered when the .NET MAUI CheckBox control is tapped, indicating that the state of the [`IsChecked`](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Buttons.SfCheckBox.html#Syncfusion_Maui_Buttons_SfCheckBox_IsChecked) property in the Checkbox control is about to change. The `StateChangingEventArgs` provides the following properties:

File: MAUI/CheckBox/Events.md [hunk 10]
```diff
@@ -55,4 +55,38 @@ This event occurs when the value or state of the [`IsChecked`](https://help.sync
...
+
+* [`IsChecked`](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Buttons.SfCheckBox.html#Syncfusion_Maui_Buttons_SfCheckBox_IsChecked) : The new value(state) of the [`IsChecked`](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Buttons.SfCheckBox.html#Syncfusion_Maui_Buttons_SfCheckBox_IsChecked) property.
+* `Cancel` : Gets or sets a value indicating whether the event should be canceled.
+
+{% tabs %}
+{% highlight xaml %}
+
+    <syncfusion:SfCheckBox x:Name="checkBox" Text="CheckBox" StateChanging="CheckBox_StateChanging"/>
```
Comment (naveenkumar-sanjeevirayan) [inline MAUI/CheckBox/Events.md:70]: set method name to OnStateChanging

File: MAUI/CheckBox/Events.md (hunk 9 above)
Comment (KarthikaRavichandranSF4617) [reply MAUI/CheckBox/Events.md]: Highlighted the event and class name.

File: MAUI/Badge-View/Badge-customization.md (hunk 7 above)
Comment (KarthikaRavichandranSF4617) [reply MAUI/Badge-View/Badge-customization.md]: Updated this description.

File: MAUI/Badge-View/Badge-customization.md (hunk 8 above)
Comment (KarthikaRavichandranSF4617) [reply MAUI/Badge-View/Badge-customization.md]: Changed the button control tag.

File: MAUI/CheckBox/Events.md [hunk 11]
```diff
@@ -55,4 +55,38 @@ This event occurs when the value or state of the [`IsChecked`](https://help.sync
 {% endhighlight %}
 {% endtabs %}
 
-![.NET MAUI CheckBox](Images/Getting-Started/tristate.png)
\ No newline at end of file
+![.NET MAUI CheckBox](Images/Getting-Started/tristate.png)
+
+## StateChanging event
+
+The `StateChanging` event is triggered when the state of the [IsChecked](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Buttons.SfCheckBox.html#Syncfusion_Maui_Buttons_SfCheckBox_IsChecked) property is about to change, either by tapping the CheckBox control or by programmatically setting the IsChecked value using XAML or C# code. The event arguments are of type `StateChangingEventArgs` and provide the following properties:
```
Comment (naveenkumar-sanjeevirayan) [inline MAUI/CheckBox/Events.md:62]: This event only triggers when change the state by tapping.

Comment (KarthikaRavichandranSF4617) [reply MAUI/CheckBox/Events.md]: Updated the content.

**Review by naveenkumar-sanjeevirayan — APPROVED** [review]

**Pull request merged**